R/transactions.R

```diff
@@ -43,4 +57,29 @@ setGeneric("dbCommit",
 setGeneric("dbRollback",
   def = function(conn, ...) standardGeneric("dbRollback"),
   valueClass = "logical"
-)
\ No newline at end of file
+)
+
+#' @aliases withTransaction,DBIConnection-method
+#' @export
+#' @rdname transactions
+setGeneric("withTransaction",
+  def = function(conn, statement, ...) standardGeneric("withTransaction"),
+  valueClass = "logical"
+)
+
+#' @export
+setMethod("withTransaction", "DBIConnection", function(conn, statement, ...) {
```

Comment:
I'd prefer an unevaluated expression, see https://github.com/rstats-db/DBI/issues/104#issuecomment-218976532 for a sketch.